### PR TITLE
viz: Add HA Option through CLI

### PR DESCRIPTION
**Pull request by Pothulapati**
This PR adds `--ha` flag for `viz install` which overrides with
the `values-ha.yaml` of the viz chart. This PR adds these functions
in `pkg/charts` so that the same can be re-used elsewhere.

## Testing

```bash
tarun in dev in on  k3d-deep () linkerd2 on  tarun/viz-ha-nits [$?] via 🐹 v1.15.4 took 2s
❯ ./bin/go-run cli viz install | grep 1024

tarun in dev in on  k3d-deep () linkerd2 on  tarun/viz-ha-nits [$?] via 🐹 v1.15.4 took 2s
❯ ./bin/go-run cli viz install --ha | grep 1024
            memory: "1024Mi"

tarun in dev in on  k3d-deep () linkerd2 on  tarun/viz-ha-nits [$?] via 🐹 v1.15.4 took 2s
❯ ./bin/go-run cli viz install --ha --set grafana.resources.memory.limit=1023Mi | grep 1023
            memory: "1023Mi"
```

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: pkg/charts/charts.go [hunk 1]
```diff
@@ -175,3 +175,41 @@ func InsertVersionValues(values chartutil.Values) (chartutil.Values, error) {
...
+	var valuesOverrideMap map[string]interface{}
+	err := yaml.Unmarshal(valuesOverride.Data, &valuesOverrideMap)
+	if err != nil {
+		return nil, err
+	}
+	return MergeMaps(valuesOverrideMap, values), nil
+}
+
+// MergeMaps returns the merge result of two maps
+func MergeMaps(a, b map[string]interface{}) map[string]interface{} {
```
Comment (adleong) [inline pkg/charts/charts.go:198]: we use mergo elsewhere in the project.  is that usable here?

Comment (Pothulapati) [reply pkg/charts/charts.go:198]: My reasons for choosing this over mergo:
- Mergo does not work semantically correct with bools. See https://github.com/imdario/mergo/issues/24 (obv due to another go reason)
- We had this same function previously too, but then this wasn't needed after the chart rendering with helm v3 refactor. This is also needed at other places too like [for merge during rendering](https://github.com/linkerd/linkerd2/pull/5433/files#diff-25fa6400f74d9645bcb61a83df4efb9f15d3da78281489c30fcff927cc62cb1dR179). so having a re-usable `mergeMap` which is semantically correct made sense.

File: pkg/charts/charts.go [hunk 2]
```diff
@@ -175,3 +175,41 @@ func InsertVersionValues(values chartutil.Values) (chartutil.Values, error) {
...
+				if bv, ok := bv.(map[string]interface{}); ok {
+					out[k] = MergeMaps(bv, v)
+					continue
+				}
+			}
+		}
+		out[k] = v
+	}
+	return out
+}
```
[the remark this answers is not in the document]
Comment (Pothulapati) [reply pkg/charts/charts.go:215]: Yes 👍

File: pkg/charts/charts.go [hunk 3]
```diff
@@ -175,3 +175,41 @@ func InsertVersionValues(values chartutil.Values) (chartutil.Values, error) {
 	}
 	return chartutil.ReadValues(InsertVersion([]byte(raw)))
 }
+
+// OverrideFromFile overrides the given map with the given file from FS
+func OverrideFromFile(values map[string]interface{}, fs http.FileSystem, chartName, name string) (map[string]interface{}, error) {
+	// Load Values file
+	valuesOverride := loader.BufferedFile{
+		Name: "values-ha.yaml",
```
[the remark this answers is not in the document]
Comment (Pothulapati) [reply pkg/charts/charts.go:183]: good find!

**Review by kleimkuhler — APPROVED** [review]
👍

**Review by cpretzer — APPROVED** [review]
LGTM, all tests passed locally

**Pull request merged**